include/oneapi/dpl/experimental/kt/esimd_radix_sort.h

```diff
@@ -0,0 +1,84 @@
+// -*- C++ -*-
+//===-- esimd_radix_sort.h --------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===------------------------------------------------------===//
+
+#ifndef _ONEDPL_KT_ESIMD_RADIX_SORT_H
+#define _ONEDPL_KT_ESIMD_RADIX_SORT_H
+
+#include "kernel_param.h"
+#include "../../pstl/utils_ranges.h"
+#include "../../pstl/hetero/dpcpp/utils_ranges_sycl.h"
+#include "../../pstl/hetero/dpcpp/parallel_backend_sycl_utils.h"
+
+#include <cstdint>
+
+#include "internal/esimd_radix_sort_utils.h"
+#include "internal/esimd_radix_sort_dispatchers.h"
+
+namespace oneapi::dpl::experimental::kt::esimd
+{
+
+// TODO: make sure to provide sufficient diagnostic if input does not allow either reading or writing
+template <bool __is_ascending = true, ::std::uint8_t __radix_bits = 8, typename _KernelParam, typename _KeysRng>
+sycl::event
+radix_sort(sycl::queue __q, _KeysRng&& __keys_rng, _KernelParam __param = {})
+{
+    if (__keys_rng.size() < 2)
+        return {};
+
+    auto __pack = __impl::__utils::__rng_pack{oneapi::dpl::__ranges::views::all(::std::forward<_KeysRng>(__keys_rng))};
```

Comment:
What is a pack?
Why does the pack contain just one element? 
Even, it is reasonable, I guess to have a comment  here  would be nice.

Reply:
This is an abstraction which stores either keys or key-value pairs. Perhaps, `std::tuple` suits here better as noted by Sergey. Let me think about it.

Reply:
I assume you looked at representing these as `zip_iterators`, for which we have `__get_sycl_range` overloads.  If there is not something which blocks it from being possible, it may help in the discussion of the repeated `size < 2` checks mentioned elsewhere by delaying `__get_sycl_range` until further in after the "packing".

Reply:
I left a not about this suggestion in #1311 for further code quality improvements